### ISSUE TITLE
Request parameter is empty: accountID

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#2782]
## Bug Description

[One user in the support forums ](https://wordpress.org/support/topic/request-parameter-is-empty-accountid-2/#post-14028273)is encountering the below error when connecting the Analytics module, specifically after selecting an account:
`Request parameter is empty: accountID`

This has been occurring with a subdomain setup only. 

Console errors when issue occurs:
```
jquery-migrate.min.js?ver=3.3.2:2 JQMIGRATE: Migrate is installed, version 3.3.2
jquery.min.js?ver=3.5.1:2 jQuery.Deferred exception: jQuery(...).live is not a function TypeError: jQuery(...).live is not a function
    at HTMLDocument.<anonymous> (https://th.cnx-software.com/wp-content/plugins/bwp-recent-comments/js/bwp-rc-widget.js?ver=5.6.1:1:69)
    at e (https://th.cnx-software.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:30005)
    at t (https://th.cnx-software.com/wp-includes/js/jquery/jquery.min.js?ver=3.5.1:2:30307) undefined
S.Deferred.exceptionHook @ jquery.min.js?ver=3.5.1:2
jquery.min.js?ver=3.5.1:2 Uncaught TypeError: jQuery(...).live is not a function
    at HTMLDocument.<anonymous> (bwp-rc-widget.js?ver=5.6.1:1)
    at e (jquery.min.js?ver=3.5.1:2)
    at t (jquery.min.js?ver=3.5.1:2)
/wp-json/google-site-kit/v1/modules/analytics/data/properties-profiles?accountID=2022395&_locale=user:1 Failed to load resource: the server responded with a status of 400 ()
googlesitekit-api.c3305021ff3630c5df8d.js:1 Google Site Kit API Error Objectcode: "missing_required_param"data: status: 400__proto__: constructor: ƒ Object()hasOwnProperty: ƒ hasOwnProperty()isPrototypeOf: ƒ isPrototypeOf()propertyIsEnumerable: ƒ propertyIsEnumerable()toLocaleString: ƒ toLocaleString()toString: ƒ toString()valueOf: ƒ valueOf()__defineGetter__: ƒ __defineGetter__()__defineSetter__: ƒ __defineSetter__()__lookupGetter__: ƒ __lookupGetter__()__lookupSetter__: ƒ __lookupSetter__()get __proto__: ƒ __proto__()set __proto__: ƒ __proto__()message: "Request parameter is empty: accountID."__proto__: constructor: ƒ Object()hasOwnProperty: ƒ hasOwnProperty()isPrototypeOf: ƒ isPrototypeOf()propertyIsEnumerable: ƒ propertyIsEnumerable()toLocaleString: ƒ toLocaleString()toString: ƒ toString()valueOf: ƒ valueOf()__defineGetter__: ƒ __defineGetter__()__defineSetter__: ƒ __defineSetter__()__lookupGetter__: ƒ __lookupGetter__()__lookupSetter__: ƒ __lookupSetter__()get __proto__: ƒ __proto__()set __proto__: ƒ __proto__()
(anonymous) @ googlesitekit-api.c3305021ff3630c5df8d.js:1
```

## Steps to reproduce

1. Setup Site Kit on a subdomain site (having previously setup Site Kit successfully on parent domain)
2. Attempt to connect the Analytics module - choosing an existing account in the dropdown during setup
3. Error appears

## Screenshots

The screenshot below shows what occurs only once an account is selected (a snapshot from [this video](https://www.youtube.com/watch?v=7gxNGnIAWG0&feature=youtu.be) kindly provided by the impacted user)

![image](https://user-images.githubusercontent.com/41326532/109818290-bf9adc80-7c2a-11eb-9124-06d41f115421.png)

## Additional Context

 - Site URL: https://th.cnx-software.com
 - [Site Health information provided](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit#response=ACYDBNiMlX6Hwz5hMWxsgwzs4zzfvbJjbQqoihe9GneHNFTwwaE9VaLdsVtjVDsTnLGgBbw)
 - No issues with parent domain
 - Only occurs with some Analytics accounts and not others
 - User has full administrator access at Analytics level
 - Occurs with no plugins active other than Site Kit via Health Check & Troubleshooting plugin (evident from video above).
 - [One previous similar topic](https://wordpress.org/support/topic/analytics-not-connected/) opened, possibly a conflict with another Analytics plugin
 - [Previous similar issue](https://wordpress.org/support/topic/request-parameter-is-empty-accountid/) reported, also on NGinx server, since resolved
 - No existing Analytics snippet exists on site

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by cnxsoft** [comment]
I've noticed it fails with two accounts, but works with another one. See screenshot below.
![connect-service](https://user-images.githubusercontent.com/1366138/107511814-90b7ba80-6bd8-11eb-8224-118e5b640818.png)

It fails with "CNX Software" and "lampang-info" accounts, but works with "CNX Software - Noticias de Sistemas Embedidos"

**Comment by jamesozzie** [comment]
Thanks for sharing, and apologies for the delayed response. Can you share the following:

- On the accounts whereby you encounter this notice do you have multiple properties assigned? If so do you recall approximately the amount of properties per account?
- When you access the Analytics web interface can you check the account settings and ensure you have [full administrator access ](https://i.imgur.com/nMwp32l.png)for those specific accounts?

**Comment by cnxsoft** [comment]
I have seven properties for CNX Software, and one property on lampang-info and "CNX Software - Noticias de Sistemas Embedidos".

I have full administrator access for CNX Software and "CNX Software - Noticias de Sistemas Embedidos", but I only have limited access to lampang-info.

**Comment by jamesozzie** [comment]
Thanks for the update. In that case if you could share a screen recording of this behavior that would be great, this is the first such occurrence and we can then try to recreate or investigate this further. 

When performing the same actions that lead to this notice please use the "CNX Software" Analytics, which you have full access to. When recording also please have your [Chrome Developer Tools open](https://i.imgur.com/jiTHddI.png), showing any console errors along the way. You can use a service such as [Loom](https://www.loom.com/) or [RecordIt](recordit.co) to easily share. 

Feel free to use the [same form](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/) previously used if you'd like to share privately.

**Comment by cnxsoft** [comment]
Here's the link to the screencast: https://youtu.be/7NacuESJgCY

**Comment by jamesozzie** [comment]
Thanks for sharing. I was able to recreate some of the console errors you encountered when using the Better WordPress Recent Comments plugin, which hasn't been updated in 7 years. I can't be sure if this is impacting Site Kit Analytics connection in your case, I've been unable to recreate the same notice. 

```
Uncaught TypeError: jQuery(...).live is not a function
    at HTMLDocument.<anonymous> (bwp-rc-widget.js?ver=5.6.2:2)
    at e (load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.6.2:2)
    at t (load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.6.2:2)
```

Have you tried following the [troubleshooting steps suggested](https://wordpress.org/support/topic/request-parameter-is-empty-accountid-2/#post-14028273) back in your WordPress support topic?

**Comment by cnxsoft** [comment]
Sorry, I missed those instructions. I've recorded the steps:
https://youtu.be/7gxNGnIAWG0

It still does not work with other plugins disabled.

**Comment by cnxsoft** [comment]
I've now selected "create a new property" or similar wording in Google Site Kit. It went through and I can see the GA code on my website, but Site Kit reports "Request parameter is empty: metrics", and no data shows in Google Analytics after several days. I'm not sure this is related or a completely different issue.

**Comment by jamesozzie** [comment]
Many thanks for sharing the additional video, which was great to see what's occurring. With this I'm going to escalate this issue while also updating the primary details. 

In relation to the additional error message I wonder is there something at Analytics account level impacting this behavior, considering this doesn't occur with all accounts. The following checks may be of use:
- Do you have any [filters](https://i.imgur.com/kCB7y5X.png) applied at Analytics account level?
- Does the same issue occur if you provide access to this Analytics account to another Google account and then sign into your WordPress site using another administrator account. When setting up Site Kit from this additional WordPress administrator user check does the same issue occur if using this new google account with this Analytics property.

**Comment by cnxsoft** [comment]
There are no filters.
I've tried with another admin account and the same issue occurs.
I've also connected PageSpeed to see what happens, and another message shows up:
"Error: Request parameter is empty: strategy."

**Comment by abdullah1908** [comment]
@jamesozzie I am unable to reproduce this issue by following the steps with the same property & account on the subdomain.

Please see screenshots for main domain & subdomain.
![parent-domain-analytics-properties](https://user-images.githubusercontent.com/14175927/109998944-8f8b3080-7d33-11eb-94e6-7f7201b17b63.png)
![subdomin-analytics](https://user-images.githubusercontent.com/14175927/109998954-9154f400-7d33-11eb-8d25-9720ccdb30cf.png)

**Comment by cnxsoft** [comment]
My main domain (cnx-software.com) works with Site Kit and the same account, but it's on another server with 4GB RAM with Ubuntu 18.04 32-bit.
My current server for th.cnx-software.com has 512MB RAM with Ubuntu 20.04 64-bit. Could this be related?

**Comment by abdullah1908** [comment]
@cnxsoft,

Thanks for updating about server stacks. Would you tell us which hosting service your are using?

We are still working on it & will let you know once we have some useful stuff to share.

Feel free to share if you find something new with the troubleshooting activity.

**Comment by cnxsoft** [comment]
@abdullah1908 I'm using Linode.

**Comment by abdullah1908** [comment]
Thanks for getting back to us. 

We'll surely look into this issue again & will get back to you soon with some insights about this issue.
Happy Monday!

**Comment by bethanylang** [comment]
@cnxsoft We're escalating this issue for further testing and we'll be in touch with any further questions and updates. Thanks again for all of your help with troubleshooting to this point!

**Comment by adamsilverstein** [comment]
Hi @cnxsoft - thanks for your help debugging this issue. I would love a little more help troubleshooting if you have a chance:

Can you take a look at your debugger network panel and let me know what you see when you select the accounts that cause the error?

In particular, look for a request to the rest api to the `/wp-json/google-site-kit/v1/modules/analytics/data/properties-profiles` URL. In the headers tab, what do you see for query string parameters?

![image](https://user-images.githubusercontent.com/2676022/111510489-71e69f80-8713-11eb-8c57-2c8ebe426821.png)

This should include the account ID, does it or is it blank? anything different about the accounts that work vs those that do not? I'm wondering if something about your account ids such as special characters might be breaking our parameter validation.

**Comment by cnxsoft** [comment]
@adamsilverstein Afer creating a new property, I was able to login, but missing parameters still happen. Please find see a screenshot showing this type of error with the metrics parameter. 
![google-site-kit-json](https://user-images.githubusercontent.com/1366138/111556576-80899100-87bd-11eb-89c1-743a3185bc52.png)

**Comment by adamsilverstein** [comment]
Hmm, this error is a little different "Parameter is empty: metrics".  This might be related to the new widget here.

Can you check under Site Kit -> Settings -> analytics, and try selecting one of the accounts that was giving you the "accountId missing" error before. Just selecting the account should trigger the error, you don't need to save the settings. I'm still curious about the query parameters there. Thanks again!

**Comment by cnxsoft** [comment]
I had to disconnect to try another account, but here's the error with accountID missing.

![Screenshot from 2021-03-19 09-18-09](https://user-images.githubusercontent.com/1366138/111722300-2a3b5180-8894-11eb-9707-e0606c52ce99.png)

**Comment by aaemnnosttv** [comment]
@adamsilverstein the query parameters are clearly there, but the errors indicate that the server doesn't recognize them.

@cnxsoft can you try putting Cloudflare into [development mode](https://support.cloudflare.com/hc/en-us/articles/200168246-Understanding-Cloudflare-Development-Mode) temporarily to test, just to make sure there isn't something potentially interfering with the request there?

**Comment by cnxsoft** [comment]
@aaemnnosttv Development mode would be a problem in this domain, as I have other live websites on the same domain, and one of my servers would not be able to handle the load. So I've just bypassed Cloudflare on th.cnx-software.com.  I suppose it's almost the same.

I still have problems accessing Analytics data through Site Kit.

By the way, there's one error I have in Site Health that does not show on my working site:
"The REST API is one way WordPress, and other applications, communicate with the server. One example is the block editor screen, which relies on this to display, and save, your posts and pages.

The REST API did not process the context query parameter correctly."

Could this be in any way related?

**Comment by aaemnnosttv** [comment]
> The REST API did not process the context query parameter correctly.

@cnxsoft yes, that sounds like the same problem indeed. From what I can tell, it seems like query parameters are not being handled properly by the server, or are being stripped somewhere along the way (server/nginx config, CDN, security filters, etc).

**Comment by cnxsoft** [comment]
@aaemnnosttv Yes, that was it. I compared my nginx configuration files, and in the websites with the issue, it was configured as follows:

` location / {
                # This is cool because no php is touched for static content
                try_files $uri $uri/ /index.php;
        `}`

I change it to
`location / {
                # This is cool because no php is touched for static content
                try_files $uri $uri/ /index.php?$args;
        }
`
and the issue is now solved. Sorry about the troubles, and thanks for your help.

**Comment by aaemnnosttv** [comment]
Thanks for sharing @cnxsoft – that would definitely do it, glad things are back to normal for you 👍 

Closing this one out as a non-issue for the plugin.
